OPAL/tac/src/main/scala/org/opalj/tac/cg/android/AndroidICCAnalysis.scala

```diff
@@ -0,0 +1,1467 @@
+
+            if (fullLifecycle) {
+                val l = ListBuffer.empty[Method]
+                for (index ← lifecycleListOption.indices) {
+                    if (lifecycleListOption(index).isEmpty) {
+                        l += aLifecycleList(index)
+                    } else l += lifecycleListOption(index).get
+                }
+                if (restart.isEmpty) {
+                    restart = Option(aLifecycleList(7))
+                }
+                val lifecycleList = l.toList
+                for (index ← 1 until lifecycleList.size) {
+                    edges += (lifecycleList(index - 1), 0) -> lifecycleList(index)
+                }
+                makeEdge(lifecycleList(3), lifecycleList(2))
+                makeEdge(lifecycleList(4), restart.get)
+                makeEdge(restart.get, lifecycleList(1))
+                makeEdge(lifecycleList(4), lifecycleList.head)
+
+            } else {
+                val lifecycleList = lifecycleListOption.flatten
+                for (index ← 1 until lifecycleList.size) {
+                    edges += (lifecycleList(index - 1), 0) -> lifecycleList(index)
+                }
+                makeEdge(pause, resume)
+                makeEdge(stop, restart)
+                makeEdge(restart, start)
+                makeEdge(stop, create)
+            }
+
+        }
+        componentMap(service).foreach { s ⇒
+            val create = s.findMethod(onCreate).headOption
+            val startComm = s.findMethod(onStartCommand).headOption
+            val bind = s.findMethod(onBind).headOption
+            val unbind = s.findMethod(onUnbind).headOption
+            val destroy = s.findMethod(onDestroy).headOption
+
+            if (fullLifecycle) {
+                val lifecycleList1 = List(create, startComm, bind, unbind, destroy)
+                val l = ListBuffer.empty[Method]
+                for (index ← lifecycleList1.indices) {
+                    if (lifecycleList1(index).isEmpty) {
+                        l += sLifecycleList(index)
+                    } else l += lifecycleList1(index).get
+                }
+                val lifecycleList = l.toList
+                makeEdge(lifecycleList.head, lifecycleList(1))
+                makeEdge(lifecycleList.head, lifecycleList(2))
+                makeEdge(lifecycleList(1), lifecycleList(4))
+                makeEdge(lifecycleList(2), lifecycleList(3))
+                makeEdge(lifecycleList(3), lifecycleList(4))
+                makeEdge(lifecycleList(1), lifecycleList(4))
+            } else {
+                makeEdge(create, startComm)
+                makeEdge(create, bind)
+                makeEdge(startComm, destroy)
+                makeEdge(bind, unbind)
+                makeEdge(unbind, destroy)
+                makeEdge(create, destroy)
+            }
+        }
+    }
+
+    def makeEdge(
+        org: Method,
```

Comment:
I'd prefer it if you used more common terminology. These are call edges? Then `src`/`source` and `tgt`/`target` are common.